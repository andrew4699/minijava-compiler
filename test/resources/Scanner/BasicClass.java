class A {
  A() {
    this.b = new B();
  }
}

class B extends A {
  B() {
    this.x = 5;
  }
}
