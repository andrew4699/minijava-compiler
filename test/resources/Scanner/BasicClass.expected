CLASS ID(A) LBRACE
  ID(A) LPAREN RPAREN LBRACE
    THIS DOT ID(b) BECOMES NEW ID(B) LPAREN RPAREN SEMICOLON
  RBRACE
RBRACE

CLASS ID(B) EXTENDS ID(A) LBRACE
  ID(B) LPAREN RPAREN LBRACE
    THIS DOT ID(x) BECOMES INT(5) SEMICOLON
  RBRACE
RBRACE
